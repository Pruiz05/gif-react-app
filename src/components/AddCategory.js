import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

     const [inputValue, setInputValue] = useState('')

     const handleInputChange = (e) =>{
        setInputValue(e.target.value)

        
     }

     const handleSubmit =(e)=>{
        e.preventDefault();

        if( inputValue.trim().length > 1){

           setCategories(c => [...c, inputValue])
           setInputValue('')
        }
      //   console.log('agregado');
      //   console.log(setCategories())
     }

    return (
        <form onSubmit={handleSubmit}>
           <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}/>
           {/* <button className="btn">Agregar</button> */}
        </form>
    )
}

// propiedad es requerida
AddCategory.propTypes = {
   setCategories: PropTypes.func.isRequired
}

