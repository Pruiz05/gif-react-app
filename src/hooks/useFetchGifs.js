import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loaing: true,
  });

  useEffect(() => {
    getGifs(category).then((imgs) => {
      setState({
        data: imgs,
        loaing: false,
      });
    });
  }, [category]);

  return state;
};

export default useFetchGifs;
