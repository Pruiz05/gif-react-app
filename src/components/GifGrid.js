import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";

import { getGifs } from "../helpers/getGifs";
import useFetchGifs from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);

  const { data: images, loading } = useFetchGifs(category);

  // useEffect(() => {
  //   getGifs(category)
  //       .then(setImages)
  // }, [category]);

  return (
    <>
      <h3 className={"animate__animated animate__fadeIn"}> {category} </h3>

      {loading && <p className={"animate__animated animate__flash"}>Loading</p>}

      <div className={"card-grid"}>
        {images.map((img) => (
          //   return <li key={img.id}>{img.title}</li>;
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes ={
  category: PropTypes.func.isRequired
}
