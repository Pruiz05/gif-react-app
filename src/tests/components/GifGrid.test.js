import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componente <GifGrid />", () => {
  const category = "One Punch";

  test("Debe de mostrarse correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category="" />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan las imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "http://localhost/test.jpg",
        title: "Prueba",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
