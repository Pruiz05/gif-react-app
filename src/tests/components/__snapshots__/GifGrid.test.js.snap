// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> debe de mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Prueba"
      url="http://localhost/test.jpg"
    />
  </div>
</Fragment>
`;
